pdfrw/objects/pdfstring.py

```diff
@@ -1,74 +1,507 @@
+
+    # These constants are used for the mode parameter to encode()
+
+    ENCODE_PDFDOC = 1   # Use PDFDocEncoding to encode Unicode
+    ENCODE_UTF16 = 2    # Use UTF-16BE to encode Unicode
+    ENCODE_LITERAL = 4  # Use () literal strings for byte encoding
+    ENCODE_HEX = 8      # Use <> hex strings for byte encoding
+
+    # Internal values used by encode
+
+    bytes_bom = codecs.BOM_UTF16_BE
+    uni_bom = codecs.BOM_UTF16_BE.decode('latin-1')
+    uni_type = type(uni_bom)
+    escape_splitter = None  # Calculated on first use
+
+    @classmethod
+    def init_escapes(cls):
+        """ Initialize the escape_splitter for the encode method
+        """
+        cls.escape_splitter = re.compile(br'(\(|\\|\))').split
+        return cls.escape_splitter
+
+    @classmethod
+    def encode(cls, source, mode=0):
+        """ The encode() constructor is called to encode a source string
+            or raw bytes into a PdfString that is suitable for inclusion
+            in a PDF.
+
+            If encode() is passed bytes (bytes type on Python3, regular
+            string on Python2), then it will skip the encoding stage
+            from Unicode to bytes.  If it is passed a Unicode string,
+            then it will do that encoding first.
+
+            NOTE 1:  There is no magic in the encoding process.  A user
+            can certainly do his own encoding, and simply initialize a
+            PdfString() instance with his encoded string.  That may be
+            useful, for example, to add line breaks to make it easier
+            to load PDFs into editors, or to not bother to escape balanced
+            parentheses, or to escape additional characters to make a PDF
+            more readable in a file editor.  Those are features not
+            supported by this method.
+
+            encode() can use a heuristic to figure out the best
+            encoding for the string, or the user can control the process
+            by setting bits in the mode parameter:
+
+                bit 0 -- force Unicode conversion using PDFDocEncoding
+                bit 1 -- force Unicode conversion using UTF-16BE
+                bit 2 -- force byte encoding using literal strings
+                bit 3 -- force byte encoding using hexadecimal strings
+
+            NOTE 2:  All other mode bits should be set to 0 for possible
+            future capability expansion.
+
+            The function does not guarantee results if the user stupidly
+            sets conflicting mode bits, but it does guarantee that it
+            will follow the instructions or raise an exception if it
+            cannot, if the user has not created a conflict.
+        """
+        raw = source
+        if isinstance(source, cls.uni_type):
+            # Give preference to pdfdocencoding, since it only
+            # requires one raw byte per character, rather than two.
+            if not (mode & cls.ENCODE_UTF16) and not source.startswith(cls.uni_bom):
```

Comment:
It's not guaranteed that I'm thinking straight, but here's what I'm trying to accomplish.  If the string starts with that code point pair, then the encoding to PDFDocEncoding will work, but will be wrong, because the decoder will incorrectly assume that it was encoded with UTF-16BE.

But if the line starts with that single unicode code point, as with any other character that cannot be represented in PDFDocEncoding, the encoding attempt should fail and we will drop back to UTF-16BE.